core/web/resolver/job_proposal_spec.go

```diff
@@ -0,0 +1,259 @@
+package resolver
+
+import (
+	"github.com/graph-gophers/graphql-go"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/services/feeds"
+)
+
+// SpecStatus defines the enum values for GQL
+type SpecStatus string
+
+const (
+	// revive:disable
+	SpecStatusPending   SpecStatus = "PENDING"
+	SpecStatusApproved  SpecStatus = "APPROVED"
+	SpecStatusRejected  SpecStatus = "REJECTED"
+	SpecStatusCancelled SpecStatus = "CANCELLED"
+	// revive:enable
+)
+
+// ToSpecStatus converts the feeds status into the enum value.
+func ToSpecStatus(s feeds.SpecStatus) (SpecStatus, error) {
+	switch s {
+	case feeds.SpecStatusApproved:
+		return SpecStatusApproved, nil
+	case feeds.SpecStatusPending:
+		return SpecStatusPending, nil
+	case feeds.SpecStatusRejected:
+		return SpecStatusRejected, nil
+	case feeds.SpecStatusCancelled:
+		return SpecStatusCancelled, nil
+	default:
+		return "", errors.New("invalid spec status")
+	}
+}
+
+// JobProposalSpecResolver resolves the Job Proposal Spec type.
+type JobProposalSpecResolver struct {
+	spec *feeds.JobProposalSpec
+}
+
+// NewJobProposalSpec creates a new JobProposalSpecResolver.
+func NewJobProposalSpec(spec *feeds.JobProposalSpec) *JobProposalSpecResolver {
+	return &JobProposalSpecResolver{spec: spec}
+}
+
+// NewJobProposalSpecs creates a slice of JobProposalSpecResolvers.
+func NewJobProposalSpecs(specs []feeds.JobProposalSpec) []*JobProposalSpecResolver {
+	var resolvers []*JobProposalSpecResolver
+
+	for i := range specs {
+		resolvers = append(resolvers, NewJobProposalSpec(&specs[i]))
+	}
+
+	return resolvers
+}
+
+// ID resolves to the job proposal spec ID
+func (r *JobProposalSpecResolver) ID() graphql.ID {
+	return int64GQLID(r.spec.ID)
+}
+
+// Definition resolves to the job proposal spec definition
+func (r *JobProposalSpecResolver) Definition() string {
+	return r.spec.Definition
+}
+
+// Version resolves to the job proposal spec version
+func (r *JobProposalSpecResolver) Version() int32 {
+	return r.spec.Version
+}
+
+// Status resolves to the job proposal spec's status
+func (r *JobProposalSpecResolver) Status() SpecStatus {
+	if status, err := ToSpecStatus(r.spec.Status); err == nil {
+		return status
+	}
+
+	return ""
```

Comment:
hm a bit tricky here, but could there be a way to not swallow the error, but at least print it in some way? Also, returning empty string as SpecStatus seems a bit strange